### PR TITLE
Add system_instruction support for lm_eval in OH

### PR DESCRIPTION
**Pull request by srajabos**
# What does this PR do?
Add system_instruction for lm_eval in OH that helps to boost accuracy for mmlu tasks

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by srajabos** [comment]
@mandy-li @skavulya - kindly review

File: examples/text-generation/run_lm_eval.py [hunk 1]
```diff
@@ -110,7 +116,8 @@ def main() -> None:
     with HabanaGenerationTime() as timer:
         with torch.no_grad():
             log_samples = args.log_samples
-            results = evaluator.simple_evaluate(lm, tasks=args.tasks, limit=args.limit_iters, log_samples=log_samples)
+            system_instruction = args.system_instruction
```
Comment (mandy-li) [inline examples/text-generation/run_lm_eval.py]: i don't see the need for line 119. For other parameters, we directly pass from args, why not system_instruction parameter?

**Review by mandy-li — COMMENTED** [review]
comment for minor change

File: examples/text-generation/run_lm_eval.py (hunk 1 above)
Comment (srajabos) [reply examples/text-generation/run_lm_eval.py]: Just followed log_samples (line 118). I'll directly pass the args.
